### ISSUE TITLE
[Bug] test not_null: error_if not working

### ISSUE THREAD

**Issue by pato0301** [dbt-labs/dbt-core#4051]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have a test within a model that has the following syntax:
- name: column name
   description: description
   tests:
     - not_null:
           - error_if: '>1'
           
This test was working okay in the 0.20 version but when I upgraded to the 0.21 version when I run `dbt test` it fails with the following error:
`Got 1 result, configured to fail if  != 0`

### Expected Behavior

The expected behaviour is that the test only fails when there is more than one null value in the model, that is to say, that the test allows only one null value.

### Steps To Reproduce

1. Have a model with a source that has a filed with one null value
2. Have a test for `not_null`
3. Add the `error_if: '>1'`  to that test
4. run `dbt test`
5. If you have none or 1 null value in that column the test should pass, else if you have more than 1 null, the test should throw an error.

### Relevant log output

```shell
Got 1 result, configured to fail if  != 0
```

### Environment

```markdown
- OS:macOS Big Sur
- Python: 3.8.8
- dbt: 0.21.0
```

### What database are you using dbt with?

bigquery

**Comment by jtcohen6** [comment]
@pato0301 Thanks for the bug report! This does indeed seem to be a regression. dbt is doing something very weird:

```yml
version: 2
models:
  - name: my_model
    columns:
      - name: id
        tests:
          - not_null:
              error_if: '>1'
```

Then in `manifest.json`:
```
      "config": {
        "enabled": true,
        "alias": null,
        "schema": "dbt_test__audit",
        "database": null,
        "tags": [],
        "meta": {},
        "materialized": "test",
        "severity": "ERROR",
        "store_failures": null,
        "where": null,
        "limit": null,
        "fail_calc": "count(*)",
        "warn_if": "!= 0",
        "error_if": "!= 0",
        "error_id": ">1"
      },
```

`error_id` ????

I'll do some more digging and try to figure out where this is happening

**Comment by willycerezo** [comment]
Do you have an ETA of fix this error? thanks in advance and regards!

**Comment by jtcohen6** [comment]
@willycerezo This fix will be included in v0.21.1 and v1.0.0. Both are currently available as release candidates:
```
pip install dbt-<adapter>==0.21.1rc1
pip install dbt-<adapter>==1.0.0rc1
```
We're likely to have a final version of v0.21.1 in the next week or two. The final v1 release will be in early December.
